lib/change_stream.js

```diff
@@ -284,11 +284,13 @@ class ChangeStreamCursor extends Cursor {
       if (this.options[optionName]) result[optionName] = this.options[optionName];
     }
 
+    const resumeKey = this.options.startAfter && !this.hasReceived ? 'startAfter' : 'resumeAfter';
```

Comment:
This looks a little suspicious to me, can you explain what's going on here?

Reply:
If a change stream started with with `startAfter` tries to resume before it has received any changes, it needs to resume with `startAfter` rather than `resumeAfter`, because the `resumeToken` is from an `invalidate` event. Without this change, test #17 fails with the error:

`Uncaught MongoError: Attempting to resume a change stream using 'resumeAfter' is not allowed from an invalidate notification.`